File: src/entity.js
const entity_directions = {
	UP: 0,
	DOWN: 1,
	LEFT: 2,
	RIGHT: 3,
	NONE: 4
};

class entity {
	direction;
	speed;
	object;
	
	constructor(
		red, green, blue,
		direction,
		width, height,
		x, y,
		speed,
		visible
	) {
		this.direction = direction;
		this.speed = speed;
		this.object = new object(
			red, green, blue,
			width, height,
			x, y,
			visible
		);
		return;
	}
	
	draw() {
		if (this.object.visible) {
			if (this.direction != entity_directions.NONE) {
				this.object.position[
					this.direction >= entity_directions.LEFT ? 0 : 1
				] += (
					this.direction % 2
				) == 0 ? -this.speed : this.speed;
			}
			this.object.draw();
		}
		return;
	}
};

File: src/entity.cs
using static impl;
using static class_object;

class class_entity : class_object {
	public enum entity_directions {
		UP,
		DOWN,
		LEFT,
		RIGHT,
		NONE
	};
	
	public entity_directions direction = entity_directions.NONE;
	public int speed = 0;
	
	public class_entity(
		byte red, byte green, byte blue,
		entity_directions direction,
		int width, int height,
		int x, int y,
		int speed,
		bool visible
	) : base(
		red, green ,blue,
		width, height,
		x, y,
		visible
	) {
		this.direction = direction;
		this.speed = speed;
		return;
	}
	
	public override void draw() {
		if (this.visible) {
			if (this.direction != entity_directions.NONE) {
				this.position[
					this.direction >= entity_directions.LEFT ? 0 : 1
				] += (
					(int)this.direction % 2
				) == 0 ? -this.speed : this.speed;
			}
			
			impl.drawFillRect(
				this.color[0], this.color[1], this.color[2],
				this.position[0], this.position[1],
				this.size[0], this.size[1]
			);
		}
		return;
	}
}